### ISSUE TITLE
Add a sidebar to the Chiventure wiki

### ISSUE THREAD

**Issue by adinilfeld** [uchicago-cs/chiventure#963]
The Chiventure [wiki page](https://github.com/uchicago-cs/chiventure/wiki) currently has the default "Pages" sidebar, which just lists all 85 wiki pages alphabetically. It would be much more organized to have a "table of contents" in the sidebar, which lists different areas of Chiventure development, along with links to the related pages. For example:

- Battles
    - [Battle Systems Existing Code Review](https://github.com/uchicago-cs/chiventure/wiki)
    - [RPG Battle System Wishlist](https://github.com/uchicago-cs/chiventure/wiki)
    - (etc.)    
- DSL
    - [DSL User Stories](https://github.com/uchicago-cs/chiventure/wiki)
    - [DSL Grammar](https://github.com/uchicago-cs/chiventure/wiki)

This way each team could quickly locate relevant wiki pages. 

I haven't figured out exactly how to do this, though [this StackOverflow thread](https://stackoverflow.com/questions/9239588/toc-or-sidebar-in-github-wiki) may be helpful.

**Comment by adinilfeld** [comment]
It seems like the simplest way of implementing a custom sidebar will be writing it in Markdown, and then using Github's "Add a custom sidebar" feature (as described in [this StackOverflow response](https://stackoverflow.com/a/25304429/13408521)). 

To test this approach, I filled the sidebar with some dummy links, which seemed to work well; I then deleted the test sidebar, and I'll work now on writing out the full table of contents in Markdown.

**Comment by adinilfeld** [comment]
After starting to write out the sidebar in Markdown, I realized how painstaking it would be to update it every time a new wiki page was added, a title changed, etc. So I'm looking into a semi-automated alternative, [github-wiki-sidebar](https://github.com/adriantanasa/github-wiki-sidebar), to see if that would be a good fit with Chiventure.

**Comment by adinilfeld** [comment]
I ran the github-wiki-sidebar program (see above comment for link), and it seemed to work well, automatically creating a sidebar with an indented list of the wiki pages.

The naming scheme is "category: page_title". For example, if the wiki page title is "Battles: Design Document", then the sidebar will show "Design Document" as an indented bullet point under "Battles."

Now I just have to update all of the wiki pages to use the above format and run the program again. Then I'll also write some documentation, so that future students know how to re-generate the sidebar whenever the wiki changes.

**Comment by adinilfeld** [comment]
As I was renaming the wiki pages, I realized that using a single `:` as the section separator was problematic, because then wiki page titles wouldn't be able to include a colon. (This would be especially annoying since dashes are already not allowed in wiki titles.) So I changed the separator to a double-colon (`::`), and I'll be going back and updating the wiki page names.

**Comment by adinilfeld** [comment]
After thinking it over, I decided that a `::` separator isn't ideal, because page titles like "Battles:: User stories" just look kind of ugly. I settled on using `||` instead: it's uncommon enough that people won't accidentally add it to their titles, and it also looks decent, e.g. "Battles || User Stories." 

So I'll go back and update the titles (again).

**Comment by adinilfeld** [comment]
Well, turns out the `|` character isn't allowed as a separator. The ones allowed are `a-z:#~ @_!`. I think out of those options, a tilde (`~`) has the best combination of being readable (e.g. Battles ~ User Stories), while also being unlikely to be used by accident in a title. So I'll be going with that, though it can always be changed later on (albeit with a lot of manual work to change each title one-by-one).

**Comment by adinilfeld** [comment]
The task is now complete. I edited all the wiki titles to use the `~` category separator, and then used [github-wiki-sidebar](https://github.com/adriantanasa/github-wiki-sidebar) to create a nice organized sidebar on the right. I also added instructions in the wiki [home page](https://github.com/uchicago-cs/chiventure/wiki) for how to update the sidebar. 

__Important__: The sidebar does _not_ update automatically when new wiki pages are added, or their titles changed. In this case, github-wiki-sidebar has to be run again (once the program is installed, it only takes a single terminal command). I emphasized this on the wiki home page -- see there for more details.

**Comment by adinilfeld** [comment]
Brief update: it seems like someone accidentally wiped the whole wiki while running the github-wiki-sidebar program, so the instructions to do so have been removed from the wiki homepage. 

In their place, I have added a brief description of the wiki naming convention (in case we use github-wiki-sidebar in the future), as well as instructions for manually adding a page to the sidebar. These can be found [here](https://github.com/uchicago-cs/chiventure/wiki).

At Borja's request, I have also added an issue (#1074) to the backlog related to creating a more complete style guide for the wiki.
